### PR TITLE
Add GitHub Actions build pipeline

### PR DESCRIPTION
**Pull request by StyxenZ**
I was curious about the build process around GW Launcher, which culminated in producing a GitHub Actions pipeline.

I'm not sure what the contribution policy is for this project, so no worries if these changes aren't required.

This pipeline runs on master and any PRs targeting master. It could be used as a status check to ensure code compiles as part of the PR process, and can be used to fully automate the release process if there's enough enthusiasm to do so.

## Changes
- Added GitHub Actions build pipeline
  - Pipeline versions and builds GW Launcher
  - Publishes the compiled artifact if building off `master`
- Added [GitVersion](https://gitversion.net/) to facilitate automatically versioning GW Launcher
- Updated `TargetFramework` to match currently produced executable
- Removed duplicate entry from `.gitignore` file

Might need to be careful around the artifact retention if we're worried about [usage limits](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes).  The build artifacts are ~50MB compressed.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# GW Launcher Development
+## GitHub Actions Build Pipeline
+GW Launcher build pipeline can be found [here](../.github/workflows/build.yml).
+
+Steps:
+- Calculate version using [GitVersion](https://gitversion.net/) based on the Git tag
+- Build GW Launcher using the `FrameworkDependent` [publish profile](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/visual-studio-publish-profiles)
```
Comment (DubbleClick) [inline docs/development.md:7]: should probably rename this

File: GitVersion.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+mode: ContinuousDelivery
+branches: {}
+tag-prefix: '[rR]'
```
Comment (DubbleClick) [inline GitVersion.yml:3]: how do you tag a new major version here? I presume it would go from 15.2 to 15.3 to 15.3, etc.

manually tag the PR?

**Review by DubbleClick — CHANGES_REQUESTED** [review]
contributions to all of our projects are generally welcome, as long as they make sense. thank you for the PR, will merge when you can answer the question above :)

File: GitVersion.yml (hunk 2 above)
Comment (StyxenZ) [reply GitVersion.yml:3]: With how things are [currently configured](https://gitversion.net/docs/reference/modes/continuous-delivery) the version is calculated based on the Git tags [here](https://github.com/gwdevhub/gwlauncher/tags). The version should be incremented each time a new tag is created. When I [ran the pipeline](https://github.com/StyxenZ/gwlauncher/actions/runs/9822898199/job/27120120430) on a fork of master, the version produced was `15.2.1`.

File: docs/development.md (hunk 1 above)
Comment (StyxenZ) [reply docs/development.md:7]: Happy to make that change as part of this PR - do you have any preferences around the naming?

File: GitVersion.yml (hunk 2 above)
Comment (StyxenZ) [reply GitVersion.yml:3]: I believe the addition of `increment: Minor` to the `GitVersion.yml` file would produce `15.3` for consistency with the existing releases. At the moment it looks like it's incrementing the patch version, if that's an issue.

Comment (DubbleClick) [reply GitVersion.yml:3]: Hmm, I'll have to check how to keep following the current versioning scheme then. Major bumps for new features or changes, minor bumps for bug fixes or small features.

File: docs/development.md (hunk 1 above)
Comment (DubbleClick) [reply docs/development.md:7]: anything that makes sense, no specific preference

File: GitVersion.yml (hunk 2 above)
Comment (DubbleClick) [reply GitVersion.yml:3]: > I believe the addition of `increment: Minor` to the `GitVersion.yml` file would produce `15.3` for consistency with the existing releases. At the moment it looks like it's incrementing the patch version, if that's an issue.

Yes, please change this to increment: Minor

New Major versions can be tagged by committing with the text `+semver: major`

File: docs/development.md (hunk 1 above)
Comment (StyxenZ) [reply docs/development.md]: That sounds descriptive enough. Updated accordingly!

File: GitVersion.yml (hunk 2 above)
Comment (StyxenZ) [reply GitVersion.yml:3]: Thanks, updated.

**Review by DubbleClick — APPROVED** [review]

**Pull request merged**